python/cuml/test/test_tsne.py

```diff
@@ -107,3 +106,30 @@ def test_tsne_large(nrows, ncols):
     Y = tsne.fit_transform(X)
     nans = np.sum(np.isnan(Y))
     assert nans == 0
+
+
+@pytest.mark.parametrize('name', dataset_names)
+@pytest.mark.parametrize('type_knn_graph', ['sklearn', 'cuml'])
+def test_tsne_knn_parameters(name, type_knn_graph):
+
+    datasets
+    X = eval("datasets.load_{}".format(name))().data
+
+    neigh = skKNN(n_neighbors=90) if type_knn_graph == 'sklearn' \
+        else cuKNN(n_neighbors=90)
+
+    neigh.fit(X)
```

Comment:
I just thought of a test that we probably should have done for UMAP. It would be nice to verify the knn graph made its way through by passing in an array other than X, which has the same shape but whose values contain absolute garbage (like all ones or zeros). Maybe something like `X_garbage`. That way, we can test the trustworthiness against the original X. If the trustworthiness drops significantly, we know the algorithm must have built the KNN on `X_garbage`. This will be good make sure future changes to the C++ don't cause issues for the Python later.

Reply:
This is a great idea, I've added this to t-SNE.